analysis/pt.lean

```diff
@@ -0,0 +1,278 @@
+/-
+Copyright (c) 2018 Koundinya Vajjha. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Koundinya Vajjha
+
+A formalization of Basic probability theory + Bayes theorem. 
+-/
+import analysis.measure_theory.measure_space tactic.tidy 
+open set measurable_space measure_theory ennreal
+
+universe u
+
+variables {α : Type u}
+
+def samp_space := Type u 
+
+@[reducible] def events (s:samp_space) := set s
+variable s:samp_space
+
+section setthry
+variables a b : events s
+
+lemma union_comm (a b : events s) : a ∪ b = b ∪ a := 
+is_commutative.comm _ a b 
+
+lemma inter_comm (a b : events s) : a ∩ b = b ∩ a := 
+is_commutative.comm _ a b 
+
+lemma inter_assoc (a b c : events s) : (a ∩ b) ∩ c = a ∩ (b ∩ c) := 
+is_associative.assoc _ a b c 
+
+lemma inter_distrib_left (a b c : events s) : (a ∩ (b ∪ c)) = (a ∩ b) ∪ (a ∩ c) := 
+inter_distrib_left a b c 
+
+lemma union_distrib_left (a b c : events s) : (a ∪ (b ∩ c)) = (a ∪ b) ∩ (a ∪ c) := 
+union_distrib_left a b c 
+
+lemma demorgan_union ( a b : events s) : -(a ∪ b) = -a ∩ -b :=
+compl_union a b 
+
+lemma demorgan_inter ( a b : events s) : -(a ∩ b) = -a ∪ -b :=
+compl_inter a b 
+
+def are_disjoint (a b : events s) := a ∩ b = ∅
+
+def are_disj_ctble {s : samp_space} (a : set(events s)) :=
+(countable a) ∧ (∀ x y ∈ a, (x ≠ y) → (x ∩ y = ∅)) 
+
+def are_disj_ctble' (f : ℕ → events s) := ∀ i j : ℕ, (i ≠ j) → (f i) ∩ (f j) = ∅ 
+
+def is_partition {s : samp_space} (a : set(events _)) := 
+(are_disj_ctble a) ∧ ( ⋃₀ a = @univ s)
+
+def is_partition' {s : samp_space} (f : ℕ  → events s) := 
+(are_disj_ctble' _ f) ∧ ((⋃i, f i) = univ) 
+
+end setthry
+
+section axfndns
+
+structure prob  (α : Type*) [measurable_space α] extends measure_theory.measure α :=
+(is_one : measure_of univ = 1)
+
+instance prob.has_coe_to_fun {α} [measurable_space α] : has_coe_to_fun (prob α) :=
+⟨λ _, set α → ennreal, λ m, m.to_measure⟩
+
+@[simp] lemma prob_eq_coerc {s : samp_space} [measurable_space s] (p : prob s) (a : set s): 
```

Comment:
I think it would make sense to move
`{s : samp_space} [measurable_space s] (p : prob s)` to a `variables` line, because it occurs in almost every statement in the rest of the file.